src/Widgets/ConnectionDialog.vala

```diff
@@ -290,6 +292,11 @@ public class Sequeler.Widgets.ConnectionDialog : Gtk.Dialog {
 		ssh_grid.attach (ssh_port_label, 0, 5, 1, 1);
 		ssh_grid.attach (ssh_port_entry, 1, 5, 1, 1);
 
+		ssh_identity_file_label = new Sequeler.Partials.LabelForm(_("SSH Identity"));
+		ssh_identity_file_entry = new Gtk.FileChooserButton (_("Select Your Identity File\u2026"), Gtk.FileChooserAction.OPEN);
```

Comment:
I'm not sure I agree with this.
Most users have no idea where do find their SSH identity file and the app should select it automatically.
I'd agree to have an extra option to select a different one, but the default should remain the expected location for unix systems.

Reply:
Indeed! I also missed that the "missing SSH key" infobar wasn't updating on selection change.

I also added added a check for empty ssh key file in ConnectionManager  but I'm not sure if that's enough to prevent old configs from breaking.

Reply:
Thanks for this, I tested with the old config and it doesn't work unless I save the config again.
I think we can avoid this by adding an extra condition to see if that key exists in the array. If it doesn't, it means is an old connection and it should use the default location.
If you can update that, than this is good to be merged, thank you so much.

Reply:
Yeah, it seems like Gee.HashMap returns null when a key is not found, which circumvents the check for empty string. That should be fixed now :+1: 

Thank you for this amazing tool